spark/src/main/java/org/opensearch/sql/spark/execution/statement/StatementModel.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.execution.statement;
+
+import static org.opensearch.sql.spark.execution.statement.StatementState.WAITING;
+
+import java.io.IOException;
+import lombok.Builder;
+import lombok.Data;
+import lombok.SneakyThrows;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.core.xcontent.XContentParserUtils;
+import org.opensearch.index.seqno.SequenceNumbers;
+import org.opensearch.sql.spark.execution.session.SessionId;
+import org.opensearch.sql.spark.execution.statestore.StateModel;
+import org.opensearch.sql.spark.rest.model.LangType;
+
+/** Statement data in flint.ql.sessions index. */
+@Data
+@Builder
+public class StatementModel extends StateModel {
+  public static final String VERSION = "version";
+  public static final String TYPE = "type";
+  public static final String STATEMENT_STATE = "state";
+  public static final String STATEMENT_ID = "statementId";
+  public static final String SESSION_ID = "sessionId";
+  public static final String LANG = "lang";
+  public static final String QUERY = "query";
+  public static final String QUERY_ID = "queryId";
+  public static final String SUBMIT_TIME = "submitTime";
+  public static final String ERROR = "error";
+  public static final String UNKNOWN = "unknown";
+  public static final String STATEMENT_DOC_TYPE = "statement";
+
```

Comment:
Can we add application id? Spark queries application id to make sure new queries will be sent to new application when the old application is terminated.

Reply:
Could query session instead? applicationId is associated with session.